src/components/Editor/Tab.js

```diff
@@ -0,0 +1,217 @@
+import React, { PureComponent } from "react";
+
+import { showMenu, buildMenu } from "devtools-contextmenu";
+
+import CloseButton from "../shared/Button/Close";
+
+import type { List } from "immutable";
+import type { SourceRecord } from "../../reducers/sources";
+
+import { getFilename, getFileURL, isPretty } from "../../utils/source";
+import { copyToTheClipboard } from "../../utils/clipboard";
+import classnames from "classnames";
+
+type SourcesList = List<SourceRecord>;
+
+type Props = {
+  tabSources: SourcesList,
+  selectSource: Object => void,
+  selectedSource: SourceRecord,
+  closeTab: string => void,
+  closeTabs: (List<string>) => void,
+  togglePrettyPrint: string => void,
+  showSource: string => void,
+  source: SourceRecord,
+  sourceTabsMetaData: {
+    [key: string]: SourceMetaDataMap
+  }
+};
+
+export default class Tab extends PureComponent<Props, State> {
+  constructor() {
```

Comment:
If we have a constructor you should take the props argument and pass it to super. But in this case since the body is empty save for the super call we should remove the constructor.